iroh-sync/src/sync.rs

```diff
@@ -532,14 +566,12 @@ impl Entry {
 /// The indentifier of a record.
 #[derive(Default, Clone, Serialize, Deserialize)]
 pub struct RecordIdentifier {
```

Comment:
Any Vec<u8> with a length >= 64 should be castable to a RecordIdentifier, right? We are not doing any checks other than the length. So I think RecordIdentifier can be done differently. Basically an unsized type.

But... next PR.